Core/NoOpRunner.Core/PlatformsContainer.cs

```diff
@@ -16,12 +16,17 @@ public PlatformsContainer(int sizeX, int sizeY) : base(sizeX, sizeY)
 
         public override void ShiftShapes()
         {
-            Shapes.ForEach(x => x.ShiftBlocks()); //Push and remove out of bounds
+            foreach (BaseShape shape in Shapes)
+                shape.ShiftBlocks(); //Push and remove out of bounds
         }
 
         public List<List<ShapeBlock>> GetNextBlocks()
         {
-            return Shapes.Select(x => x.GetNextBlocks()).ToList();
+            List<List<ShapeBlock>> result = new List<List<ShapeBlock>>();
+            foreach (BaseShape shape in Shapes)
+                result.Add(shape.GetNextBlocks());
```

Comment:
They'll want at least one place where you use the iterator methods explicitly

Reply:
Changed to Shapes.GetEnumerator(); and while loop